apps/admin_app/assets/js/views/product/edit.js

```diff
@@ -35,6 +36,23 @@ export function imageOnEnter() {
   });
 }
 
+export function onToggleActivation() {
+  $(document).delegate('.variant_activate', 'change', function (event) {
+    var id = $(this).find('.variant_id').val();
+    var target_element = $(this);
+    var state = ( $(this).find('.switch-input').prop("checked") == true) ? "active" : "draft";
+    var data = {state: state, id: id};
+    $.ajax({
+      url: `/variant_state`,
+      type: "GET",
```

Comment:
Should the method request be `PATCH`?

Reply:
Ideally yes, if it deals with updating just specific field in a struct.